components/ai_chat/core/browser/engine/engine_consumer_conversation_api.cc

```diff
@@ -0,0 +1,156 @@
+
+void EngineConsumerConversationAPI::ClearAllQueries() {
+  api_->ClearAllQueries();
+}
+
+void EngineConsumerConversationAPI::GenerateRewriteSuggestion(
+    std::string text,
+    const std::string& question,
+    GenerationDataCallback received_callback,
+    GenerationCompletedCallback completed_callback) {
+  std::vector<ConversationEvent> conversation = {
+      {mojom::CharacterType::HUMAN, ConversationEventType::UserText, text},
+      {mojom::CharacterType::HUMAN, ConversationEventType::RequestRewrite,
+       question}};
+  api_->PerformRequest(std::move(conversation), std::move(received_callback),
+                       std::move(completed_callback));
+}
+
+void EngineConsumerConversationAPI::GenerateQuestionSuggestions(
+    const bool& is_video,
+    const std::string& page_content,
+    SuggestedQuestionsCallback callback) {
+  std::vector<ConversationEvent> conversation{
+      GetAssociatedContentConversationEvent(page_content, is_video),
+      {mojom::CharacterType::HUMAN,
+       ConversationEventType::RequestSuggestedActions, ""}};
+
+  auto on_response = base::BindOnce(
+      &EngineConsumerConversationAPI::OnGenerateQuestionSuggestionsResponse,
+      weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  api_->PerformRequest(std::move(conversation), base::NullCallback(),
+                       std::move(on_response));
+}
+
+void EngineConsumerConversationAPI::OnGenerateQuestionSuggestionsResponse(
+    SuggestedQuestionsCallback callback,
+    GenerationResult result) {
+  if (!result.has_value() || result->empty()) {
+    // Query resulted in error
+    LOG(ERROR) << "Error getting question suggestions.";
+    std::move(callback).Run(base::unexpected(std::move(result.error())));
+    return;
+  }
+
+  // Success
+  std::vector<std::string> questions =
+      base::SplitString(*result, "|", base::WhitespaceHandling::TRIM_WHITESPACE,
+                        base::SplitResult::SPLIT_WANT_NONEMPTY);
+  std::move(callback).Run(std::move(questions));
+}
+
+void EngineConsumerConversationAPI::GenerateAssistantResponse(
+    const bool& is_video,
+    const std::string& page_content,
+    std::optional<std::string> selected_text,
+    const ConversationHistory& conversation_history,
+    const std::string& human_input,
+    GenerationDataCallback data_received_callback,
+    GenerationCompletedCallback completed_callback) {
+  std::vector<ConversationEvent> conversation;
+  // associated content
+  if (!page_content.empty()) {
+    conversation.push_back(
+        GetAssociatedContentConversationEvent(page_content, is_video));
+  }
+  // history
+  for (const auto& message : conversation_history) {
+    if (message.selected_text.has_value() && !message.selected_text->empty()) {
+      conversation.push_back({mojom::CharacterType::HUMAN,
+                              ConversationEventType::PageExcerpt,
+                              message.selected_text->c_str()});
```

Comment:
Why is c_str() needed instead of just *message.selected_text or message.selected_text.value()?